ocaml/libs/tracing/tracing.ml

```diff
@@ -642,22 +695,22 @@ module Export = struct
       try
         debug "Tracing: About to export" ;
         Hashtbl.iter
-          (fun trace_id span_list ->
+          (fun _trace_id span_list ->
             let zipkin_spans = Content.Json.Zipkinv2.content_of span_list in
             match
               match endpoint with
               | Url url ->
                   Http.export ~span_json:zipkin_spans ~url
               | Bugtool ->
-                  File.export ~trace_id ~span_json:zipkin_spans
-                    ~path:!File.trace_log_dir
+                  File.export ~span_json:zipkin_spans ~path:!File.trace_log_dir
             with
             | Ok _ ->
                 ()
             | Error e ->
                 raise e
           )
-          span_list
+          span_list ;
+        File.FileStream.close_stream ()
```

Comment:
Separating concerns leads to simpler-looking code: the export function can be defined first, and then be used for processing the spans:
```ocaml
let export = match endpoint with
  | Url url ->  Http.export ~url
  | Bugtool -> File.export ~path:!File.trace_log_dir
in
Hashtbl.iter (fun _ span ->
  let span_json = Content.Json.Zipkinv2.content_of span in
  match export ~span_json with
  | Ok _ -> ()
  | Error e -> raise e
  )
  span_list ;
```

Reply:
[c783311](https://github.com/xapi-project/xen-api/pull/5038/commits/c783311f36d5b9e362947efad4376a27b8fd74b1)